File: dotnet-core/Tests/BrowserActionLineTest.cs
using Xunit;
using System;
using System.Collections.Generic;

namespace TeaParser
{
    public class BrowserActionLineTest
    {

        [Fact]
        [Trait("Category", "Unit")]
        public void CanLoadBrowserActionLine()
        {
            var act1 = "click";
            var selector1 = "xpath";
            var locator1 = "//*[text() = 'Log In']";
            var act2 = "type";
            var selector2 = "id";
            var locator2 = "loginUserName";
            var act3 = "click";
            var selector3 = "name";
            var locator3 = "passwordField";
            var act4 = "type";
            var selector4 = "password";
            var locator4 = "//*[@id='password']";
            var list = new List<Token>(){
                new Token(Token.TokenType.TOKEN_ACTION, 0, act1),
                new Token(Token.TokenType.TOKEN_ACTION, 1, selector1),
                new Token(Token.TokenType.TOKEN_ACTION, 2, locator1),
                new Token(Token.TokenType.TOKEN_ACTION, 0, act2),
                new Token(Token.TokenType.TOKEN_ACTION, 1, selector2),
                new Token(Token.TokenType.TOKEN_ACTION, 2, locator2),
                new Token(Token.TokenType.TOKEN_ACTION, 0, act3),
                new Token(Token.TokenType.TOKEN_ACTION, 1, selector3),
                new Token(Token.TokenType.TOKEN_ACTION, 2, locator3),
                new Token(Token.TokenType.TOKEN_ACTION, 0, act4),
                new Token(Token.TokenType.TOKEN_ACTION, 1, selector4),
                new Token(Token.TokenType.TOKEN_ACTION, 2, locator4),
                new Token(Token.TokenType.TOKEN_EOE, 4, null)
            };


            var lines = new BrowserActionLine(list);
            Assert.Equal(String.Join(" ", act1, selector1, locator1), lines.ActionLines[0].ToString());
            Assert.Equal(String.Join(" ", act2, selector2, locator2), lines.ActionLines[1].ToString());
            Assert.Equal(String.Join(" ", act1, selector3, locator3), lines.ActionLines[2].ToString());
            Assert.Equal(String.Join(" ", act2, selector4, locator4), lines.ActionLines[3].ToString());
        }

        [Fact]
        [Trait("Category", "Unit")]
        public void CanInterpretATeaFileInput()
        {
            string url = "https://www.reddit.com/login/?experiment_d2x_sso_login_link=enabled";
            string tc = "Test Case: Trying to login to reddit with a random username/password will fail.";
            // string action1 = "click xpath //*[text() = 'Log In']";
            string action1 = "click xpath //*[@id='loginUsername']";
            string action2 = "type id random";
            string action3 = "click id loginPassword";
            string action4 = "type id password";
            string action5 = "click xpath //fieldset[5]/button";
            
            var tea = new TeaFile(@"./../../../Resources/LoginWithFailingUsername.tea");

            Assert.Equal(url, tea.URL);
            Assert.Equal(tc, tea.TestCase);
            
            // Assert.Equal(action1, tea.ActionLines[0].ToString());
            Assert.Equal(action1, tea.ActionLines[0].ToString());
            Assert.Equal(action2, tea.ActionLines[1].ToString());
            Assert.Equal(action3, tea.ActionLines[2].ToString());
            Assert.Equal(action4, tea.ActionLines[3].ToString());
            Assert.Equal(action5, tea.ActionLines[4].ToString());
                 
        }


    }
}

File: java/tea-framework/src/test/java/com/tea/framework/SeleniumWrapper/Drivers/BrowserActionLineTest.java
package com.tea.framework.SeleniumWrapper.Drivers;

import com.tea.framework.parser.BrowserActionLine;
import com.tea.framework.parser.TeaFile;
import com.tea.framework.parser.Token;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;

public class BrowserActionLineTest {

    @Test
    public void CanLoadBrowserActionLine() {
        String act1 = "click";
        String selector1 = "xpath";
        String locator1 = "//*[text() = 'Log In']";
        String act2 = "type";
        String selector2 = "id";
        String locator2 = "loginUserName";
        String act3 = "click";
        String selector3 = "name";
        String locator3 = "passwordField";
        String act4 = "type";
        String selector4 = "password";
        String locator4 = "//*[@id='password']";
        List<Token> list = new ArrayList<Token>() {{
            add(new Token(Token.TokenType.TOKEN_ACTION, 0, act1));
            add(new Token(Token.TokenType.TOKEN_ACTION, 1, selector1));
            add(new Token(Token.TokenType.TOKEN_ACTION, 2, locator1));
            add(new Token(Token.TokenType.TOKEN_ACTION, 0, act2));
            add(new Token(Token.TokenType.TOKEN_ACTION, 1, selector2));
            add(new Token(Token.TokenType.TOKEN_ACTION, 2, locator2));
            add(new Token(Token.TokenType.TOKEN_ACTION, 0, act3));
            add(new Token(Token.TokenType.TOKEN_ACTION, 1, selector3));
            add(new Token(Token.TokenType.TOKEN_ACTION, 2, locator3));
            add(new Token(Token.TokenType.TOKEN_ACTION, 0, act4));
            add(new Token(Token.TokenType.TOKEN_ACTION, 1, selector4));
            add(new Token(Token.TokenType.TOKEN_ACTION, 2, locator4));
            add(new Token(Token.TokenType.TOKEN_EOE, 4, null));
        }};


        BrowserActionLine lines = new BrowserActionLine(list);
        Assertions.assertEquals(String.join("\t", act1, selector1, locator1), lines.getActionLines().get(0).toString());
        Assertions.assertEquals(String.join("\t", act2, selector2, locator2), lines.getActionLines().get(1).toString());
        Assertions.assertEquals(String.join("\t", act1, selector3, locator3), lines.getActionLines().get(2).toString());
        Assertions.assertEquals(String.join("\t", act2, selector4, locator4), lines.getActionLines().get(3).toString());
    }

    @Test
    public void CanInterpretATeaFileInput() throws FileNotFoundException {
        String teaFile = "src/test/resources/LoginWithFailingUsername.tea";
        String url = "https://www.reddit.com/login/?experiment_d2x_sso_login_link=enabled";
        String tc = "Test Case: Trying to login to reddit with a random username/password will fail.";
        // String action1 = "click xpath //*[text() = 'Log In']";
        String action1 = "click\txpath\t//*[@id='loginUsername']";
        String action2 = "type\tid\trandom";
        String action3 = "click\tid\tloginPassword";
        String action4 = "type\tid\tpassword";
        String action5 = "click\txpath\t//fieldset[5]/button";

        TeaFile tea = new TeaFile(teaFile);

        Assertions.assertEquals(url, tea.getUrl());
        Assertions.assertEquals(tc, tea.getTestCase());

        // Assertions.assertEquals(action1, tea.ActionLines[0].ToString());
        Assertions.assertEquals(action1, tea.getActionLines().get(0).toString());
        Assertions.assertEquals(action2, tea.getActionLines().get(1).toString());
        Assertions.assertEquals(action3, tea.getActionLines().get(2).toString());
        Assertions.assertEquals(action4, tea.getActionLines().get(3).toString());
        Assertions.assertEquals(action5, tea.getActionLines().get(4).toString());
    }
}